### PR TITLE
New script added to change channel names

### PR DESCRIPTION
**Pull request by emilroz**
Script renames channel names for the following data types: Image, Dataset, Project, Well, Plate and Screen.

### CODE REVIEW

**Comment by joshmoore** [comment]
Hmmm....not sure why travis would be failing like this. @sbesson?

**Comment by sbesson** [comment]
See https://github.com/ome/omero-user-scripts/pull/11

File: util_scripts/Change_Channel_Names.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            counter += 1
+            image_list = self.query_service.findAllByQuery(
+                self.get_image_query, params)
+            print "Retrived %i images" % len(image_list)
+            for image in image_list:
+                check_lc = image.getPixels(0).getChannel(0). \
+                    getLogicalChannel().getId().getValue()
+                if check_lc not in lc_ids:
+                    exit = True
+                    break
```
Comment (chris-allan) [inline util_scripts/Change_Channel_Names.py]: Drop the conditional and just `return` here.

File: util_scripts/Change_Channel_Names.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                        getLogicalChannel().setName(
+                            rstring(self.new_channel_names[c]))
+                    lc_id = \
+                        image.getPixels(0).getChannel(c). \
+                        getLogicalChannel().getId().getValue()
+                    lc_ids.remove(lc_id)
+                    lc_ids_batch.remove(lc_id)
+            image_list = \
+                self.update_service.saveAndReturnArray(image_list)
+            if exit:
```
Comment (chris-allan) [inline util_scripts/Change_Channel_Names.py]: No need for this if you `return` above.

File: util_scripts/Change_Channel_Names.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        counter = 0
+        exit = False
+        while len(lc_ids_batch) > 0:
+            params.page(counter * paging, paging)
+            counter += 1
+            image_list = self.query_service.findAllByQuery(
+                self.get_image_query, params)
+            print "Retrived %i images" % len(image_list)
+            for image in image_list:
+                check_lc = image.getPixels(0).getChannel(0). \
```
Comment (chris-allan) [inline util_scripts/Change_Channel_Names.py]: You can use the convenience method `getPrimaryPixels()` in place of `getPixels(0)`.

File: util_scripts/Change_Channel_Names.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        return [lc_id.getValue() for lc_id in lc_ids]
+
+    def renameBatch(self, lc_ids, step):
+        lc_ids_batch = random.sample(lc_ids, step)
+        params = omero.sys.ParametersI()
+        params.addIds(lc_ids_batch)
+        number_of_channels = len(self.new_channel_names)
+        paging = self.image_paging
+        counter = 0
+        exit = False
```
Comment (chris-allan) [inline util_scripts/Change_Channel_Names.py]: This conditional is not required.

File: util_scripts/Change_Channel_Names.py [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+                    getLogicalChannel().getId().getValue()
+                if check_lc not in lc_ids:
+                    exit = True
+                    break
+                print "Renaming", image.getName().getValue(), \
+                    image.getId().getValue()
+                image_noc = image.getPixels(0).getSizeC().getValue()
+                if image_noc != number_of_channels:
+                    print "\tChannels don't match, skipping"
+                    for c in range(0, image_noc):
```
Comment (chris-allan) [inline util_scripts/Change_Channel_Names.py]: With `range` and `xrange` a starting point of `0` is the default and implied.

File: util_scripts/Change_Channel_Names.py [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+        scriptParams = {}
+        for key in client.getInputKeys():
+            if client.getInput(key):
+                scriptParams[key] = client.getInput(key, unwrap=True)
+
+        # wrap client to use the Blitz Gateway
+        conn = BlitzGateway(client_obj=client)
+        nameChanger = renameChannels(conn, scriptParams)
+        message = nameChanger.run()
+        #message = "DOne"
```
Comment (chris-allan) [inline util_scripts/Change_Channel_Names.py]: If this commented out code is not required, remove it.

**Pull request merged**